tempo_embeddings/text/corpus.py

```diff
@@ -60,12 +61,16 @@ def extend(self, passages: list[Passage]) -> list[int]:
         """
         start_index = len(self._passages)
         self._passages.extend(passages)
+        end_index = len(self._passages)
 
         if self._umap:
-            # Compute all UMAP embeddings with existing UMAP model
-            self._embeddings_2d = self._umap.transform(self.embeddings)
+            self.embeddings_2d = np.append(
```

Comment:
I wonder if this logic is correct. Shouldn't the UMAP take all of the points again and re-compute the total mapping to consider all point-wise relationships? I am not sure if the result is similar enough this way (but I do get that this is more efficient). Can we double-check this?

Reply:
I was considering that as well. Efficiency is one point indeed. Also, in current examples where we expand with e.g. 10 points per cluster while we already have 1000s of points in total, the new points should probably not make a big difference.
Anyway, it is hard to validate the quality of 2d embeddings. Any ideas?

An option could be to add an argument that optionally triggers a recomputation of the UMAP model when adding new passages, but I'm afraid that shifting the responsibility to the user does not solve the question of what is the best trade-off.

Reply:
@angel-daza For now, I have added more details to the docstring. I would suggest tackling this issue once we have been able to run more experiments and see if/when this is a problem.
Let me know if you approve.

Reply:
I guess a formal way to test it would be to reduce one of the classic labeled ML datasets and compare the two mechanisms...

But I think you are right, given the proportion of points, the model should be already powerful enough to transform the extra points, which by definition are near the centroid anyway, so it should not significantly affect their final location.